lib/models/addon.js

```diff
@@ -1486,6 +1506,13 @@ if (addonTreeCacheExperimentPresent) {
 
     return cacheKey; // profit?
   };
+
+  Object.defineProperty(addonProto, 'treeCache', {
```

Comment:
Lets move this to be another export instead of adding to prototype.

Something like:

```js
if (experiments.ADDON_TREE_CACHING) {
  module.exports._treeCache = ADDON_TREE_CACHE;
}
```

I definitely don't want addon authors thinking this is just a run of the mill public API...